bson-kotlin/src/test/kotlin/org/bson/codecs/kotlin/DataClassCodecTest.kt

```diff
@@ -133,6 +135,20 @@ class DataClassCodecTest {
         assertDecodesTo(withStoredNulls, dataClass)
     }
 
+    @Test
+    fun testDataClassWithListThatLastItemDefaultsToNull() {
```

Comment:
This is just a copy from `bson-kotlinx` - to ensure the `bson-kotlin` library also works as expected.